### PR TITLE
Implement calculating of RTT for down tracks

### PR DESCRIPTION
**Pull request by ashellunts**
Thanks Jie for rtt.go code

### CODE REVIEW

File: pkg/telemetry/rtt.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+	"github.com/pion/rtcp"
+)
+
+const (
+	NSUS                 = 1000
+	microssec      int64 = 1000000
+	kNtpJan1970Sec int64 = 2208988800
+)
+const kMaxCompactNtp uint32 = 0xFFFFFFFF
```
Comment (boks1971) [inline pkg/telemetry/rtt.go:15]: maybe include all these constants in the above block.

File: pkg/telemetry/rtt.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+import (
+	"math"
+	"time"
+
+	"github.com/pion/rtcp"
+)
+
+const (
+	NSUS                 = 1000
```
Comment (boks1971) [inline pkg/telemetry/rtt.go:11]: maybe more descriptive name like `nanosInMicro` (that also does not sound great, but `NSUS` is very cryptic, I am assuming it is nano seconds in a micro second)

File: pkg/telemetry/rtt.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	microssec      int64 = 1000000
```
Comment (boks1971) [inline pkg/telemetry/rtt.go:12]: Same thing here `microsInSecond` or something?

File: pkg/telemetry/rtt.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	if remain > half {
+		return uint32(quot + 1)
+	}
+	return uint32(quot)
+}
+
+// TimeMicrosToNtp convert us to NtpTime (from webrtc)
+func TimeMicrosToNtp(time_us int64) *NtpTime {
+	// Since this doesn't return a wallclock time, but only NTP representation
+	// of rtc::TimeMillis() clock, the exact offset doesn't matter.
```
Comment (boks1971) [inline pkg/telemetry/rtt.go:42]: `rtc::TimeMillis` reference, is that valid?

File: pkg/telemetry/rtt.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	// To simplify conversions between NTP and RTP time, this offset is
+	// limited to milliseconds in resolution.
+	time_ntp_us := time_us + kNtpJan1970Sec*microssec
+
+	// TODO(danilchap): Convert both seconds and fraction together using int128
```
Comment (boks1971) [inline pkg/telemetry/rtt.go:47]: Is this TODO valid?

File: pkg/telemetry/rtt.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	// when that type is easily available.
+	// Currently conversion is done separetly for seconds and fraction of a second
+	// to avoid overflow.
+
+	// Convert seconds to uint32 through uint64 for well-defined cast.
+	// Wrap around (will happen in 2036) is expected for ntp time.
+	ntp_seconds := uint32(time_ntp_us / microssec)
```
Comment (boks1971) [inline pkg/telemetry/rtt.go:54]: We use `camelCase` for variables.

File: pkg/telemetry/test/telemetry_service_test.go [hunk 7]
```diff
@@ -155,7 +155,7 @@ func Test_OnDownStreamRTCP(t *testing.T) {
 	require.Equal(t, 1, int(stats[0].NackCount))
 	require.Equal(t, 1, int(stats[0].PliCount))
 	require.Equal(t, 1, int(stats[0].FirCount))
-	require.Equal(t, 4, int(stats[0].Delay))      // max of delay, see list of rtcp.ReceptionReport above
+	require.Equal(t, 0, int(stats[0].Delay))
```
Comment (boks1971) [inline pkg/telemetry/test/telemetry_service_test.go:158]: Looks like need some extra test util to ensure there is some valid data. Makes it harder with the use of current clock. Maybe a TODO?

**Review by boks1971 — APPROVED** [review]
A few comments to adapt Jie's code to LK style.

File: pkg/telemetry/rtt.go (hunk 4 above)
Comment (cnderrauber) [reply pkg/telemetry/rtt.go:42]: these code are used by me before,  are extracted from webrtc's RR handle, convert C++ to go, I think these comment can be all removed. feel free to change this.

**Review by cnderrauber — APPROVED** [review]

File: pkg/telemetry/rtt.go (hunk 1 above)
Comment (ashellunts) [reply pkg/telemetry/rtt.go:15]: done

File: pkg/telemetry/rtt.go (hunk 2 above)
Comment (ashellunts) [reply pkg/telemetry/rtt.go:11]: done

File: pkg/telemetry/rtt.go (hunk 3 above)
Comment (ashellunts) [reply pkg/telemetry/rtt.go:12]: done

File: pkg/telemetry/rtt.go (hunk 4 above)
Comment (ashellunts) [reply pkg/telemetry/rtt.go:42]: removed

File: pkg/telemetry/rtt.go (hunk 5 above)
Comment (ashellunts) [reply pkg/telemetry/rtt.go:47]: removed

File: pkg/telemetry/test/telemetry_service_test.go (hunk 7 above)
Comment (ashellunts) [reply pkg/telemetry/test/telemetry_service_test.go:158]: added TODO

File: pkg/telemetry/rtt.go (hunk 6 above)
Comment (ashellunts) [reply pkg/telemetry/rtt.go]: done

**Review by boks1971 — APPROVED** [review]

**Pull request merged**